scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/index.js

```diff
@@ -38,6 +41,7 @@ const ingredientHandlers = {
   "data.instance_methods?": classMembers.handleDataInstanceMethods,
   "data.instance_properties?": classMembers.handleDataInstanceProperties,
   "data.interactive_example?": handleDataInteractiveExample,
+  "data.permitted_properties": handleDataPermittedProperties,
```

Comment:
It's actually `data.permitted_properties?`, which probably means that the factory needs to get and understand an `isOptional` flag.